go/vt/vtgate/evalengine/arithmetic.go

```diff
@@ -243,6 +243,7 @@ func isByteComparable(v sqltypes.Value) bool {
 		return true
 	}
 	switch v.Type() {
+	// TODO: extend to support Varchar
```

Comment:
why would arithmetic evalengine support varchar?

Reply:
To support de-duplication in union statements. L223 mentions to support hashcode for all possible types.